datastore/leveldb/leveldbds.nim

```diff
@@ -0,0 +1,157 @@
+import std/options
+import std/tables
+import std/os
+import std/strformat
+
+import pkg/leveldbstatic
+import pkg/chronos
+import pkg/questionable
+import pkg/questionable/results
+from pkg/stew/results as stewResults import isErr
+import pkg/upraises
+
+import ../datastore
+import ../datastore/defaultimpl
+
+push: {.upraises: [].}
+
+type
+  LevelDbDatastore* = ref object of Datastore
+    db: LevelDb
+    locks: TableRef[Key, AsyncLock]
+
+func toByteSeq(str: string): seq[byte] {.inline.} =
+  @(str.toOpenArrayByte(0, str.high))
+
+func toString(bytes: openArray[byte]): string {.inline.} =
+  let length = bytes.len
+  if length > 0:
+    result = newString(length)
+    copyMem(result.cstring, bytes[0].unsafeAddr, length)
+
+method has*(self: LevelDbDatastore, key: Key): Future[?!bool] {.async, locks: "unknown".} =
```

Comment:
I think that you can remove the `locks` pragma, since the implementation below is known and you know that there are locks used.